### DISCUSSION TITLE
Natural Scroll and Tap-Click Emulation

### DISCUSSION THREAD

**Discussion by Infinidoge** [qtile/qtile#2447] [Q&A] [answered]
Looking into Qtile as a GNOME user on Arch Linux, currently on a laptop. Something I've been looking into, but haven't found an answer for yet, is how to enable natural scrolling (might also just be inverted scrolling) and emulating clicks by tapping with multiple fingers (1 finger tap for left click, 2 finger tap for right click, etc.).

Is there a known or best way to do both of these within the Qtile config?

**Comment by m-col** [comment, marked as answer]
Standalone window managers like Qtile generally don't get involved in these kinds of settings.

I'm pretty sure the scroll setting would be set within a given application, e.g. I have a setting in Firefox for this. If you're after replicating the setting within GNOME's settings GUI then I think that only applies to GTK applications. I think this is configured outside of GNOME with `dconf`; this page looks promising: https://bbs.archlinux.org/viewtopic.php?id=179667

For tap-to-click settings this is handled by the input driver for the touchpad, which is probably libinput. This is configured using X11's configs, as described here: https://wiki.archlinux.org/title/Libinput#Via_Xorg_configuration_file (the `"Tapping"` setting).

Hope that helps!

**Reply by Infinidoge** [reply]
Will go ahead and try them out! Was looking for a WM level solution since I saw i3 doing similar, but those make sense.

**Reply by Infinidoge** [reply]
Oh hey, `libinput` includes an option for Natural Scrolling, perfect!

**Reply by ramnes** [reply]
> Will go ahead and try them out! Was looking for a WM level solution since I saw i3 doing similar, but those make sense.

What is doing i3, out of curiosity? From what I remember it doesn't handle any of these either.

**Reply by Infinidoge** [reply]
> 
> 
> > Will go ahead and try them out! Was looking for a WM level solution since I saw i3 doing similar, but those make sense.
> 
> What is doing i3, out of curiosity? From what I remember it doesn't handle any of these either.

I was probably mistaken, I do remember having to configure something to get it working properly in i3, but it was likely outside of i3. I can't for the life of me remember where I configured it though.

**Reply by Etn40ff** [reply]
IIRC all of these are X11 settings, see for example `man 4 synaptics`
